### PR TITLE
Add Base.firstindex(c::Chain) = 1

### PR DESCRIPTION
**Pull request by KronosTheLate**
I see no reason not to have this, and I ran into missing it, as I have made a habit of indexing with `begin` as opposed to 1.

I am not sure - should this be added to NEWS.md? It seems too minor for it...
- [ ] Entry in NEWS.md

### CODE REVIEW

File: src/layers/basic.jl [hunk 1]
```diff
@@ -68,7 +68,7 @@ end
 Base.getindex(c::Chain, i::AbstractArray) = Chain(c.layers[i])
 Base.getindex(c::Chain{<:NamedTuple}, i::AbstractArray) =
   Chain(NamedTuple{Base.keys(c)[i]}(Tuple(c.layers)[i]))
-
+Base.firstindex(c::Chain) = 1
```
Comment (ToucheSir) [inline src/layers/basic.jl:71]:
```suggestion
```
This line shouldn't be necessary with the forwarded definition above.

Comment (KronosTheLate) [reply src/layers/basic.jl:71]: Cool! I have never seen forwarding before, but I trust you got it right.

**Comment by ToucheSir** [comment]
Do you mind adding a test or two for begin/end while you're at it? I don't believe we currently check the latter either.

**Review by ToucheSir — APPROVED** [review]
This is a nice little QoL change, thanks!

**Pull request merged**